component/src/main/java/org/wso2/extension/siddhi/io/twitter/source/TwitterSource.java

```diff
@@ -0,0 +1,539 @@
+                query = QueryBuilder.createQuery(queryParam, count, searchLang, sinceId, maxId, until, since,
+                        resultType, geocode, latitude, longitude, radius, unitName);
+                twitterConsumer.consume(twitter, query, sourceEventListener, siddhiAppContext, pollingInterval);
+            }
+        } catch (Exception e) {
+            throw new ConnectionUnavailableException(
+                    "Error in connecting with the Twitter API : " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * This method can be called when it is needed to disconnect from the end point.
+     */
+    @Override
+    public void disconnect() {
+        if (twitterStream != null) {
+            twitterStream.shutdown();
+            if (log.isDebugEnabled()) {
+                log.debug("The status listener has been cleared!");
+            }
+        }
+    }
+
+    /**
+     * Called at the end to clean all the resources consumed by the {@link Source}
+     */
+    @Override
+    public void destroy() {
+        if (twitterStream != null) {
+            twitterStream.clearListeners();
+            if (log.isDebugEnabled()) {
+                log.debug("The twitter stream has been shutdown !");
+            }
+        }
+    }
+
+    /**
+     * Called to pause event consumption
+     */
+    @Override
+    public void pause() {
+        twitterConsumer.pause();
+    }
+
+    /**
+     * Called to resume event consumption
+     */
+    @Override
+    public void resume() {
+        twitterConsumer.resume();
+    }
+
+    /**
+     * Used to collect the serializable state of the processing element, that need to be
+     * persisted for the reconstructing the element to the same state on a different point of time
+     *
+     * @return stateful objects of the processing element as a map
+     */
+    @Override
+    public Map<String, Object> currentState() {
+        Map<String, Object> currentState = new HashMap<>();
+        currentState.put(TwitterConstants.POLLING_SEARCH_SINCEID, twitterConsumer.tweetId);
+        return currentState;
+    }
+
+    /**
+     * Used to restore serialized state of the processing element, for reconstructing
+     * the element to the same state as if was on a previous point of time.
+     *
+     * @param map the stateful objects of the processing element as a map.
+     *            This map will have the  same keys that is created upon calling currentState() method.
+     */
+    @Override
+    public void restoreState(Map<String, Object> map) {
+        sinceId = Long.parseLong(map.get(TwitterConstants.POLLING_SEARCH_SINCEID).toString());
+    }
+
+    /**
+     * Used to validate the parameters.
+     */
+
+    private void validateParameter() {
+        Query.ResultType resultType1 = Query.ResultType.valueOf(this.resultType);
+        if (mode.equalsIgnoreCase(TwitterConstants.MODE_STREAMING)) {
+            for (String s : staticOptionsKeys) {
```

Comment:
Shall we rename the s to something more readable like "staticOptionKey", it should be a key right?